tests/integration/test_bayesian_optimization.py

```diff
@@ -198,9 +228,30 @@ def build_model(data: Dataset) -> GaussianProcessRegression:
         kernel.lengthscales.prior = tfp.distributions.LogNormal(
             tf.math.log(kernel.lengthscales), scale
         )
-        gpr = gpflow.models.GPR((data.query_points, data.observations), kernel, noise_variance=1e-5)
-        gpflow.utilities.set_trainable(gpr.likelihood, False)
-        return GaussianProcessRegression(gpr)
+
+        if model_type == "GPR":
+            gpr = gpflow.models.GPR(
+                (data.query_points, data.observations), kernel, noise_variance=1e-5
+            )
+            gpflow.utilities.set_trainable(gpr.likelihood, False)
+            return GaussianProcessRegression(gpr, **model_args)
+        elif model_type == "VGP":
+            likelihood = gpflow.likelihoods.Gaussian(1e-3)
+            vgp = gpflow.models.VGP(initial_data.astuple(), kernel, likelihood)
+            gpflow.utilities.set_trainable(vgp.likelihood, False)
+            return VariationalGaussianProcess(vgp, **model_args)
+        elif model_type == "SVGP":
+            Z = search_space.sample_sobol(20)  # Initialize diverse inducing locations
+            svgp = gpflow.models.SVGP(
+                kernel,
+                gpflow.likelihoods.Gaussian(variance=1e-5),
+                Z,
+                num_data=len(data.observations),
+            )
+            gpflow.utilities.set_trainable(svgp.likelihood, False)
+            return SparseVariational(svgp, BatchOptimizer(tf.optimizers.Adam(0.1)))
```

Comment:
seems like a high learning rate for adam, but if the tests pass I guess its fine